contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,114 @@
+
+        if (keccak256(abi.encodePacked(parentDomain)) != keccak256(abi.encodePacked("")) && domains[parentDomain].controller != address(0)) {
+            domains[parentDomain].childDomains.push(keccak256(abi.encodePacked(cleanDomainName)));
+        }
+
+        domains[cleanDomainName] = Domain({
+            controller: msg.sender,
+            deposit: msg.value,
+            childDomains: new bytes32[](0)
+        });
+    }
+
+    function releaseDomain(string memory domainName) public onlyController(domainName) {
+        string memory cleanDomainName = DomainValidator.stripProtocol(domainName);
+        require(domains[cleanDomainName].childDomains.length == 0,"Remove child domains first");
+        require(domains[cleanDomainName].controller != address(0),"Domain is not reserved");
+
+        removeChildFromParentDomain(cleanDomainName);
+
+        address payable controller = payable(msg.sender);
+        uint256 refundAmount = domains[cleanDomainName].deposit;
+        delete domains[cleanDomainName];
+
+        controller.transfer(refundAmount);
+    }
+
+    function removeChildFromParentDomain(string memory childDomain) private {
+        string memory parentDomain = DomainValidator.getParentDomain(childDomain);
+        bytes32 hashedChild = keccak256(abi.encodePacked(childDomain));
+        bytes32 hashedParent = keccak256(abi.encodePacked(parentDomain));
+
+        if (hashedParent != keccak256(abi.encodePacked("")) &&domains[parentDomain].controller != address(0)) {
+            bytes32[] storage childDomains = domains[parentDomain].childDomains;
+            uint index;
+            bool found = false;
+
+            for (uint i = 0; i < childDomains.length; i++) {
+                if (childDomains[i] == hashedChild) {
+                    index = i;
+                    found = true;
+                    break;
+                }
+            }
+
+            if (found) {
+                assembly {
+                    let len := sload(childDomains.slot)
+                    if lt(index, sub(len, 1)) {
+                        let data := add(add(childDomains.slot, 1), index)
+                        let lastElem := add(
+                            add(childDomains.slot, 1),
+                            sub(len, 1)
+                        )
+                        sstore(data, sload(lastElem))
+                    }
+                    sstore(childDomains.slot, sub(len, 1))
+                }
+            }
+        }
+    }
+
+    function changeDeposit(string memory domainName, uint256 newDeposit) public payable onlyController(domainName) sufficientDeposit(newDeposit) {
+        string memory cleanDomainName = DomainValidator.stripProtocol(domainName);
+        domains[cleanDomainName].deposit = msg.value;
+    }
+
+    function transferDomainControl(string memory domainName, address newController) public onlyController(domainName) {
```

Comment:
Although, not critical but I wouldn't do it or at least asked the product manager if it is something that we need